README.md

```diff
@@ -1 +1,571 @@
+
+### Field Nullability
+
+Circumstances change. What is true today may not be in the future. Considering this, it is important to understand how nullability affects schema evolution.
+
+#### Output
+
+Bias toward nullable output fields. Never make promises to clients that you cannot keep. If you tell them that something will never be `null` then you close yourself to the possibility that it might, in the future, be `null`. Prefer nullable fields so that clients are written to handle `null`s with good defaults. This is slightly more work upfront for front-end developers but provides long-term flexibility.
+
+#### Input
+
+Bias toward __non__-nullable input fields and arguments. A non-nullable field which was technically optional can become required without breaking clients. However, a nullable field cannot become non-nullable without breaking clients. Try to have every input field and argument be non-nullable.
+
+### Client-Driven
+
+Move complexity to resolvers (when it does not negatively affect UX).
+
+For example, imagine we have a DB in which we store the user's given name and surname as well as their date of birth and a UI where we show the user's name and age in milliseconds. A good schema might be as follows:
+
+```graphql
+type User {
+  fullName: String
+  dateOfBirth: ISO8601DateTime
+}
+```
+- We provide the client with `fullName`, instead of just `givenName` and `surname`. There is no advantage to the UI concatenating strings, especially when it is complicated by nullable names and potentially multiple languages, some of which have surnames are last names and some of which have surnames are first names. This is even more useful when multiple clients, built from different codebases, need to show the same data. We can provide `givenName` and `surname` fields, but we should also provide the info that clients require instead of making clients jump through hoops to render correctly.
+- Even though we are displaying the user's age in milliseconds, we don't expose this field directly. This is because, unlike `fullName`, `ageInMilliseconds` changes constantly. This is a case where the complexity of calculating the age must be done on the client. Calculating the age on the server would cause the UX to suffer.
+
+### Redundant Fields
+
+Do not fear redundant fields.
+
+There is no limit to the number of fields and resolvers your schema has. The only problem posed by additional fields is additional maintenance. This cost is minimal, especially if it allows the clients to do less work. Do not worry about adding fields, especially if it allows you to deprecate other fields.
+
+### Mutations
+
+Mutations are the most difficult part of GraphQL schema design.
+
+#### RPC, not CRUD
+
+This is the key concept. Mutations should be small and generally directly linked to an action the user is taking.
```

Comment:
Is it fair to say that a "create" mutation (like create `case`) would have the minimal number of input fields for a `case` to be valid, while there could (or should) be several "update" mutations that each accept only a single update-able field?

I am working with a schema which has an update mutation with several input fields such that when changing one of the fields, the current values for the non-changing fields must also be sent -- this is what we want to avoid, yes?